lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1336,7 +1340,69 @@ class MongoClientInterface {
     }
 
     /**
-     * internal listing function
+     * internal listing function for buckets in v0
+     * @param {String} bucketName bucket name
+     * @param {Object} params internal listing params returned by listing extension
+     * @param {Object} extension listing extention
+     * @param {Object} log logger
+     * @param {Function} cb callback
+     * @return {undefined}
+     */
+    internalListObjectV0(bucketName, params, extension, log, cb) {
+        const c = this.getCollection(bucketName);
+        const stream = new MongoReadStream(c, params, params.mongifiedSearch);
+        const skip = new Skip({
+            extension,
+            gte: params.gte,
+        });
+        let cbDone = false;
+
+        skip.setListingEndCb(() => {
+            stream.emit('end');
+            stream.destroy();
+        });
+        skip.setSkipRangeCb(range => {
+            // stop listing this key range
+            stream.destroy();
+
+            // update the new listing parameters here
+            const newParams = params;
+            newParams.start = undefined; // 'start' is deprecated
+            newParams.gt = undefined;
+            newParams.gte = range;
+
+            // then continue listing the next key range
+            this.internalListObject(bucketName, newParams, extension, log, cb);
+        });
+        stream
+            .on('data', e => {
+                skip.filter(e);
+            })
```

Comment:
```suggestion
            .on('data', entry => {
                skip.filter(entry);
            })
```

It is easier to understand with full variable names

Reply:
yep i agree